12_color.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Implementation for RubyMonk Primer Problem 12 Enough contrast?
+# Problem Statement:
+# For 2 Colors in RGB:
+# (R1, G1, B1) and (R2, G2, B2),
+# Brightness index is:
+# ( 299*R1 + 587*G1 + 114*B1) / 1000
+# Brightness difference is:
+# Absolute difference in brighness indices
+# Hue difference is:
+# |R1 - R2| + |G1 - G2| + |B1 - B2|
+# where |x| is the absolute value of x
+# If Brightness difference is more than 125 and the Hue difference is more than 500,
+# then the colors have sufficient contrast
+# Find out if the given color combos have sufficient contrast and get all the tests passing.
+class Color
+  # rubocop:disable Naming/MethodParameterName
+  attr_reader :r, :g, :b
+
+  def initialize(r, g, b)
+    @r = r
+    @g = g
+    @b = b
+  end
+
+  def brightness_index
+    # your code here
+    ((299 * r) + (587 * g) + (114 * b)) / 1000
```

Comment:
I disagree.  It's ruby, so any kind of checking for type is heavily discouraged and probably the wrong way to go.  Decimal types respond to `#*` just fine, so I'm not sure what this would gain?

Reply:
Yes, that was my thought as well.  Generally if it's an external user/caller, I will guard.  If a guard would be added, I'd put it in the CTOR which was provided by RubyMonk, so I try and not modify that.